builtin/credential/ldap/backend.go

```diff
@@ -76,82 +77,25 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		return "", nil, logical.ErrorResponse("password cannot be of zero length when passwordless binds are being denied"), nil, nil
 	}
 
-	ldapClient := ldaputil.Client{
-		Logger: b.Logger(),
-		LDAP:   ldaputil.NewLDAP(),
-	}
-
-	c, err := ldapClient.DialLDAP(cfg.ConfigEntry)
+	ldapClient, err := ldap.NewClient(ctx, ldaputil.ConvertConfig(cfg.ConfigEntry))
 	if err != nil {
 		return "", nil, logical.ErrorResponse(err.Error()), nil, nil
 	}
-	if c == nil {
-		return "", nil, logical.ErrorResponse("invalid connection returned from LDAP dial"), nil, nil
-	}
 
 	// Clean connection
-	defer c.Close()
```

Comment:
All of the code deleted below is implemented via the `Authenticate` method found in [cap/ldap](https://github.com/hashicorp/cap/blob/f28ec041b24bf50a3ce40185f5ee31053d7cb240/ldap/client.go#L217).

We do need to specify options to also get `groups` and `attributes`.